src/App.jsx

```diff
@@ -1,56 +1,128 @@
-    setColors([{ id: uid(), ...newColor }, ...colors]);
+    const updatedColor = { id: uid(), ...newColor };
+    setColors([updatedColor, ...colors]);
+    if (currentTheme.id === "t1") {
+      return;
+    } else {
+      const updatedTheme = {
+        ...currentTheme,
+        colors: [updatedColor, ...currentTheme.colors],
+      };
+      updateCurrentTheme(updatedTheme);
+    }
   }
 
   function handleDeleteColor(id) {
-    const colorsToKeep = colors.filter((color) => {
-      return color.id !== id;
-    });
-    setColors(colorsToKeep);
+    if (currentTheme.id === "t1") {
+      return;
+    } else {
+      const updatedColors = currentTheme.colors.filter(
+        (color) => color.id !== id
+      );
+      const updatedTheme = { ...currentTheme, colors: updatedColors };
+      updateCurrentTheme(updatedTheme);
+    }
   }
 
   function handleEditColor(id, newColor) {
-    const newEditColor = colors.map((color) => {
-      return color.id === id ? { ...color, ...newColor } : color;
-    });
-    setColors(newEditColor);
+    if (currentTheme.id === "t1") {
+      return;
+    } else {
+      const updatedColors = currentTheme.colors.map((color) =>
+        color.id === id ? { ...color, ...newColor } : color
+      );
+
+      const updatedTheme = { ...currentTheme, colors: updatedColors };
+      updateCurrentTheme(updatedTheme);
+    }
+  }
+
+  function handleAddTheme(newThemeName) {
+    const newTheme = { id: uid(), name: newThemeName, colors: [] };
+    setThemes([...themes, newTheme]);
+    setCurrentTheme(newTheme);
+  }
+
+  function handleChangeTheme(event) {
+    const selectedThemeId = event.target.value;
+    const selectedTheme = themes.find((theme) => theme.id === selectedThemeId);
+    setCurrentTheme(selectedTheme);
+  }
+
+  function handleEditTheme(updatedTheme) {
+    const updatedThemes = themes.map((theme) =>
+      theme.id === updatedTheme.id ? updatedTheme : theme
+    );
+
+    setThemes(updatedThemes);
+    setCurrentTheme(updatedTheme);
+  }
+
+  function handleDeleteTheme(themeId) {
+    if (currentTheme.id === "t1") {
+      return;
+    } else {
+      const updatedThemes = themes.filter((theme) => theme.id !== themeId);
+      setThemes(updatedThemes);
+
+      if (updatedThemes.length > 0) {
+        setCurrentTheme(updatedThemes[0]);
+      } else {
+        setCurrentTheme(null);
+      }
+    }
   }
 
   return (
     <div className="app">
       <h1>Theme Creator</h1>
-      <ColorForm onAddColor={handleAddColor} />
-      {colors.length === 0 ? (
-        <p className="new-color-text">
-          Please add a new color!{" "}
-          <span role="img" aria-label="rainbow emoji">
-            🌈
-          </span>
-        </p>
+      <ThemesMenu
+        themes={themes}
+        currentTheme={currentTheme}
+        onAddTheme={handleAddTheme}
+        onChange={handleChangeTheme}
+        onEditTheme={handleEditTheme}
+        onDeleteTheme={handleDeleteTheme}
```

Comment:
and if you applied the formentioned suggestions I think you should add :`isAddingTheme={isAddingTheme}

        setIsAddingTheme={setIsAddingTheme}` as props for the themes menu